### PR TITLE
Addition of Hotdog Pad

### PR DESCRIPTION
**Pull request by onnenon**
## Description
Add Hotdog Pad macropad.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

NA

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/onnenon/hotdog_pad/keyboard.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
+    "manufacturer": "onnenon",
+    "keyboard_name": "Hotdog Pad",
+    "maintainer": "onnenon",
+    "processor": "RP2040",
+    "bootloader": "rp2040",
+    "dynamic_keymap": {
+        "layer_count": 4
+    },
```
Comment (zvecr) [inline keyboards/onnenon/hotdog_pad/keyboard.json:9]:
```suggestion
```

File: keyboards/onnenon/hotdog_pad/keyboard.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        "vid": "0x4F4E"
+    },
+    "features": {
+        "caps_word": true,
+        "encoder": true,
+        "rgblight": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "combos": true
```
Comment (zvecr) [inline keyboards/onnenon/hotdog_pad/keyboard.json:25]:
```suggestion
        "encoder": true,
        "rgblight": true,
        "extrakey": true,
        "mousekey": true
```

File: keyboards/onnenon/hotdog_pad/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,18 @@
+// Copyright 2024 Stephen Onnen (@onnenon)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {[0] = LAYOUT(KC_A, KC_B, KC_C, KC_D, KC_MUTE)};
```
Comment (zvecr) [inline keyboards/onnenon/hotdog_pad/keymaps/default/keymap.c:6]:
```suggestion
const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
    [0] = LAYOUT(KC_A, KC_B, KC_C, KC_D, KC_MUTE)
};
```

File: keyboards/onnenon/hotdog_pad/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] = {ENCODER_CCW_CW(KC_VOLU, KC_VOLD)},
+};
+#endif
+
+// Enable the power pin for the Xiao Seeed rp2040 onboard NeoPixel
+void matrix_init_kb(void) {
+    gpio_set_pin_output(GP11);
+    gpio_write_pin_high(GP11);
+}
```
Comment (zvecr) [inline keyboards/onnenon/hotdog_pad/keymaps/default/keymap.c:18]: Please add a file `keyboards/onnenon/hotdog_pad/hotdog_pad.c` with the following content:
```c

// Copyright 2024 Stephen Onnen (@onnenon)
// SPDX-License-Identifier: GPL-2.0-or-later
#include "quantum.h"

void matrix_init_kb(void) {
    // Enable the power pin for the Xiao Seeed rp2040 onboard NeoPixel
    gpio_set_pin_output(GP11);
    gpio_write_pin_high(GP11);

    matrix_init_user();
}
```

And remove the implementation here
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/onnenon/hotdog_pad/keymaps/default/keymap.c (hunk 4 above)
Comment (onnenon) [reply keyboards/onnenon/hotdog_pad/keymaps/default/keymap.c]: Thanks for the guidance here. I wasn't sure of best place to put this.

File: keyboards/onnenon/hotdog_pad/hotdog_pad.c [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+#include "quantum.h"
+
+void matrix_init_kb(void) {
+    // Enable the power pin for the Xiao Seeed rp2040 onboard NeoPixel
+    gpio_set_pin_output(GP11);
+    gpio_write_pin_high(GP11);
+
+    matrix_init_user();
+}
```
Comment (zvecr) [inline keyboards/onnenon/hotdog_pad/hotdog_pad.c:12]: Sorry, forgot to update to a non-matrix callback in the previous review.

```suggestion
void keyboard_pre_init_kb(void) {
    // Enable the power pin for the Xiao Seeed rp2040 onboard NeoPixel
    gpio_set_pin_output(GP11);
    gpio_write_pin_high(GP11);

    keyboard_pre_init_user();
}
```

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**